### DISCUSSION TITLE
CheckboxList bulkToggleable defaults to "Deselect all"

### DISCUSSION THREAD

**Discussion by korematter** [filamentphp/filament#6159] [Help] [answered]
Hi there,

Finding an interesting behavior with the CheckboxList with bulkToggleable as it defaults with "Deselect all" when no options are selected. Just a simple test here as evidence:

```
CheckboxList::make('test')
    ->options(['1' => 'one', '2' => 'two'])
    ->bulkToggleable(),
```

![image](https://user-images.githubusercontent.com/58536534/230777181-6594162b-dab1-47b9-bd10-3ae3d08a5bbe.png)

Does anyone know if there is a way to default it to "Select all"?

Thanks!
Steve

**Comment by korematter** [comment, marked as answer]
Nevermind. Just upgraded to Forms 2.17.22 and it now behaves as expected.  You guys are on it! Thanks!

**Comment by abdikafar** [comment]
I am facing the same issue even after upgrading versions. although I am using the admin panel not only the froms. I on version filament V2.17.49. can you help with it?

**Reply by danharrin** [reply]
Try running `php artisan filament:upgrade`

**Reply by abdikafar** [reply]
I did that and still same issue. for context here is my code. `return $form
            ->schema([
                //class_id
                Forms\Components\Select::make('class_id')
                    ->label('Class')
                    ->columnSpan(2)
                    ->options(AcademicClass::all()->map(function ($class) {
                        return [
                            'label' => $class->name . ' ( ' . $class->teacher->name . ' )'. ' ( ' . $class->time . ' )',
                            'value' => $class->id,
                        ];
                    })->pluck('label', 'value'))
                    ->reactive()
                    ->required(),
                Forms\Components\Card::make()
                    ->schema([
                        CheckboxList::make('students')
                            ->options(function (\Closure $get) {
                                if($get('class_id')) {
                                        $class = AcademicClass::find($get('class_id'));
                                        return $class->students->pluck('name', 'id');
                                    }
                                    return [];
                                })
                            ->columnSpan(2)
                            ->label('Students')
                            ->formatStateUsing(function (   $context, $state, \Closure $get) {
                                if ($context === 'create') {
                                    if ($get('class_id')) {
                                        $class = AcademicClass::find($get('class_id'));
                                        return $class->students->pluck('name', 'id');
                                    }
                                    return [];
                                }else{
                                    return $state;
                                }
                            })
                            ->reactive()
                            ->bulkToggleable(),
                        ]),
            ]);` . here is a screenshot 
![fl](https://github.com/filamentphp/filament/assets/16723354/32c53c17-1bcf-4966-af7a-a91c4e7a5a16)

**Reply by korematter** [reply]
@abdikafar , your code is a bit unreadable so please try to format it to better help support you.  From what I can make out, not sure if you can use formatStateUsing() on a CheckboxList. This is used mainly on Tables\Columns\TextColumn. Might not fix your issue though but worth removing to try isolating the problem. Also, try using a relationship() rather than option() which is what I was using with success.
